src-tauri/src/main.rs

```diff
@@ -43,28 +43,19 @@ fn plot_progress_tracker() -> usize {
 }
 
 #[tauri::command]
-async fn farming(path: String, reward_address: String, plot_size: u64) {
-    match reward_address.len() {
-        0 => {
-            farm(path.into(), "ws://127.0.0.1:9944", None, plot_size)
-                .await
-                .unwrap();
-        }
-        _ => {
-            if let Ok(address) = parse_reward_address(&reward_address) {
-                farm(path.into(), "ws://127.0.0.1:9944", Some(address), plot_size)
-                    .await
-                    .unwrap();
-            } else {
-                error!("Reward address could not be parsed!");
-            }
-        }
+async fn farming(path: String, reward_address: String, plot_size: u64) -> [u8; 32] {
+    if let Ok(address) = parse_reward_address(&reward_address) {
+        farm(path.into(), "ws://127.0.0.1:9944", Some(address), plot_size)
+            .await
+            .unwrap()
+    } else {
+        unreachable!("Reward address could not be parsed on the backend!")
```

Comment:
`unreachable!()` is for code that never happens. This should have returned an error to the frontend instead (or at least boolean about whether operation succeeded).

Reply:
Actually this should never happen. It is a double-check. 
We are already parsing the reward address on the frontend side and ensuring it is in the correct format. 
I can as well remove the double-check on the backend side if you like

Reply:
I mean from the point of view of the backend it can definitely happen. You can't make that decision based on locally available information, hence `unreachable!()` is not appropriate.

Reply:
Ok, makes sense